velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -0,0 +1,137 @@
+
+namespace facebook::velox {
+
+class DecimalArithmeticTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testDecimalExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+    ASSERT_EQ(result->type()->toString(), expected->type()->toString());
+  }
+
+  template <typename T>
+  void testOpFlatConstant(
+      const std::string& operation,
+      const std::string& constant,
+      const VectorPtr& flatVector,
+      const VectorPtr& result,
+      bool isLeftConstant) {
+    auto shortConstant = BaseVector::wrapInConstant(
+        1, 0, makeDecimalFlatVector<ShortDecimal>({ShortDecimal(1000)}, 10, 3));
+    std::vector<VectorPtr> input({flatVector});
+    RowVectorPtr rowVector = makeRowVector(input);
+    auto rowType = std::dynamic_pointer_cast<const RowType>(rowVector->type());
+    core::TypedExprPtr typedExpr;
+    std::vector<core::TypedExprPtr> params;
+    auto fieldTypedExpr =
+        std::make_shared<core::FieldAccessTypedExpr>(flatVector->type(), "c0");
+    auto constantTypedExpr =
+        std::make_shared<core::ConstantTypedExpr>(shortConstant);
+    if (isLeftConstant) {
+      typedExpr =
+          makeTypedExpr(fmt::format("{}({},c0)", operation, constant), rowType);
+      params.push_back(constantTypedExpr);
+      params.push_back(fieldTypedExpr);
+    } else {
+      typedExpr =
+          makeTypedExpr(fmt::format("{}(c0,{})", operation, constant), rowType);
+      params.push_back(fieldTypedExpr);
+      params.push_back(constantTypedExpr);
+    }
+    auto newCallTypedExpr = std::make_shared<core::CallTypedExpr>(
+        typedExpr->type(), params, operation);
+    auto actual = evaluate(newCallTypedExpr, rowVector);
+    VectorPtr actualSimple = std::dynamic_pointer_cast<SimpleVector<T>>(actual);
+    assertEqualVectors(actualSimple, result);
+    ASSERT_EQ(result->type()->toString(), actual->type()->toString());
+  }
+
+  template <typename T>
+  void testOpDictVectors(
+      const std::string& operation,
+      const VectorPtr& result,
+      const std::vector<VectorPtr>& flatVector) {
+    // Dictionary vectors as arguments.
+    vector_size_t newSize = flatVector[0]->size() * 2;
+    auto indices = makeIndices(newSize, [&](int row) { return row / 2; });
+    auto shortDictA =
+        VectorTestBase::wrapInDictionary(indices, newSize, flatVector[0]);
+    auto shortDictB =
+        VectorTestBase::wrapInDictionary(indices, newSize, flatVector[1]);
+    auto resultDict =
+        VectorTestBase::wrapInDictionary(indices, newSize, result);
+    testDecimalExpr<T>(
+        resultDict,
+        fmt::format("{}(c0,c1)", operation),
+        {shortDictA, shortDictB});
+  }
+};
+} // namespace facebook::velox
+
+TEST_F(DecimalArithmeticTest, decimalAddTest) {
```

Comment:
drop decimal and Test: TEST_F(DecimalArithmeticTest, add) {

Reply:
Done.